### ISSUE TITLE
Changes to the HtmlDocument/HtmlElement class namespace

### ISSUE THREAD

**Issue by FilipToth** [dotnet/runtime#52498]
## Background and Motivation

I need a quick and easy way to serialize HTML and then perform LINQ operations on it and, I tried not to use external libraries. When I was researching into this, I saw that the `HtmlDocument` class is in the `System.Windows.Forms` namespace. Now, it would be awesome if we could have a way to serialize HTML without having to use the `System.Windows.Forms` namespace plus, I just don't think that it's the correct namespace to put this class in. I am not sure what other namespace we should put this API in so, I really want to hear your thoughts.

## Proposed API

```diff
namespace System.Windows.Forms
{
-    public class HtmlDocument
}

namespace System.Text.Html
{
+   public class HtmlDocument
}
```

## Alternative Designs

Again, I am not sure if the `System.Text.Html` namespace is right for this class.

## Risks

I am also given to understand that the `WebBrowser` class uses this API so, that might be an issue and the reason as to why this API is in the namespace it currently is. Other than that, I don't think there are any other risks with this.

**Comment by GrabYourPitchforks** [comment]
Can you define what you mean by "serialize HTML"? The existing `HtmlDocument` class is a wrapper around DOM and a UI-based browser window, but that doesn't quite sound like what you're looking for. Your description makes it sound like you're looking for something much more constrained.

**Comment by FilipToth** [comment]
Yes, I am just looking for a DOM wrapper and I was wondering why the `HtmlDocument` class was in the `System.Windows.Forms` namespace. There are other applications for the `HtmlDocument` class outside the WinForms environment so, why is it in that namespace?

**Comment by KalleOlaviNiemitalo** [comment]
HtmlDocument doesn't even have any public constructors; you can only get an instance from the WebBrowser control, or perhaps from a Control that is being hosted as ActiveX in HTML.

The implementation seems to depend on MSHTML, which I don't think is available for anything other than Windows. So moving this to System.Text.Html could cause developers to be disappointed when it does not work elsewhere, in addition to being a breaking change.

PowerShell used to depend on MSHTML for HTML parsing but this was disabled in the port to .NET Core (<https://github.com/PowerShell/PowerShell/pull/1468>) and later removed altogether (<https://github.com/PowerShell/PowerShell/pull/5376>).

**Comment by GrabYourPitchforks** [comment]
What are some examples of code constructs you need? Are you trying to do something like "find every \<a\> tag in the document and pull its _href_ value"?

**Comment by FilipToth** [comment]
Yes, pretty much everything the `HtmlDocument` class can do just with a proper constructor. I know there're libraries like the Html Agility Pack but, I thought a similar implementation could also be included in the runtime API.

**Comment by KalleOlaviNiemitalo** [comment]
Does Microsoft Edge WebView2 support an unattended mode that could be used here? The DevTools protocol includes commands for reading the DOM.

(I have been wondering if Microsoft Edge WebView2 could be used for HTML-to-PDF conversion in DocFX v2, which requires running JavaScript code and nowadays uses an old fork of WebKit via wkhtmltopdf. However, according to <https://github.com/MicrosoftEdge/WebView2Feedback/issues/364>, Microsoft Edge WebView2 does not support unattended export to PDF. I don't know whether that applies to unattended use in general.)

**Comment by FilipToth** [comment]
I can't answer your question because I don't have experience with WebView2 but, WebView2 is not a part of the runtime API. I can always use say the HTML Agility Pack or some library. I was just wondering if we’re planning on adding something like this into the runtime.
